### ISSUE TITLE
Discussion: Switch to openSCAD or other open modelling software?

### ISSUE THREAD

**Issue by DocSunset** [probatio/probatio#12]
Should we switch to an open source 3D modeller such as openSCAD, freeCAD, solvespace, or another?

### Fusion360

Pros:

- excellent user interface and user experience (generally)
- very powerful!
- handles multi-part assemblies very well, e.g. including downloaded STEP file models of buttons/sensors
- familiar to several of us
- existing models already finished in fusion360

Cons:

- proprietary software
- full power is behind a pay wall
- mild vendor lock in
- poor compatibility with git version control and collaboration workflow
    - in designs without any external links (e.g. an IGES model imported into your design), you can use [a workflow where you import f3d files to open and export to save as](https://www.reddit.com/r/Fusion360/comments/gyyoee/is_it_possible_to_use_git_with_fusion_360/fteugta/)
    - in designs that contain external references, [it is practically impossible to use git version control and collaboration features.](https://github.com/probatio/probatio/issues/14#issue-842888798)
- documentation mainly in the form of forum posts and seminar videos (ugh)
- automatic updates, forced planned obsolescence
- all the above lead to barriers for "just anyone" to participate
- ... proprietary software... >:(

### General open source

Pros:

- free, libre, open source, human rights respecting
- generally good git workflow integration
- documentation generally better than fusion360 (imo)
- easier for newcomers to participate in development without having to pay money or make an account with a private corporation

Cons:

- generally less polished user experience
- generally not as powerful as easily (may take more effort to achieve similar results)
- may require us to learn new tools
- would have to re-create many existing models

### OpenSCAD

Pros:

- design is code; super duper extra parametric!
- super duper extra git compatible
- possibility to provide thingiverse-style customizer user experience
- great UI/UX for programmers

Cons:

- poor UI/UX for non-programmers
- very different design approach; will require some hard re-implementation of existing fusion360 models
- only basic support for assemblies via model-in-place paradigm (body = part)
    - this means e.g. no simulation of mechanical linkage movements that you get with constraint-based assemblies, not that we are using any such features at the moment from what I can tell?
- doesn't support many 3D model file formats, may be difficult to import part models for illustrative assembly models

### FreeCAD

Pros:

- similar constraint-based parametric modelling approach as Fusion360; relatively simple to reimplement existing models?
- near feature-parity with Fusion360 through easily installed add-ons

Cons:

- UI/UX leaves something to be desired

### OpenSCAD *and* FreeCAD

Pros:

- as described above
- using both packages covers up most cons of either
- use the right tool for the job!

Cons:

- using multiple design tools increases barrier to entry for newcomers, who have to learn two modelling tools
- higher maintenance burden to coordinate models and design workflow across two packages

**Comment by DocSunset** [comment]
I will be exploring the git-compatibility of fusion360's exported design files. If things integrate with git more smoothly than I am expecting, I will adjust the pro-con list in the first post.

**Comment by jtragtenberg** [comment]
I am now committing some f3d and f3z files from Fusion 360 I designed. This f3z can include external components, like 3d modes of the buttons, potentiometers that we can download from the internet in step files. Is it easy to do that in the open source alternatives? They are really handy to communicate to the maker what components to put use with that 3d model

**Comment by DocSunset** [comment]
Yeah, handling multi-part assemblies is something fusion360 excels at. I believe similar functionality is possible in freeCAD using add-ons, but I don't think any open source packages can handle this sort of thing out of the box. I'll add mention of this to the pro-con list above.

**Comment by DocSunset** [comment]
Added some more analysis of specific open source packages to the first comment.
